app/README.md

```diff
@@ -0,0 +1,105 @@
+# App modules
+
+App modules are subsections of the main site, dealing with a specific area , e.g locations or moves. They are constructed using Express routers.
+
+They can have a variety of different parts, in either a root file, or a folder with multiple files, and exposed via an `index.js` in the folder.
+
+- controllers
+  - Express
+- constants
+  - Predefined default settings for a module, including routing paths, actions, and other defaults
+- fields
+  - Form-wizard field definitions
+- middleware
+  - Express middleware for the module, rendering against the views
+- steps
+  - Form-wizard step definitions
+- validators
+  - Form-wizard field validator functions
+- views
+  - Nunjucks templates
+- index (routing)
+  - Main entrypoint for the module, tying together all the parts of the module in the form of an Express router
+
+## Index / Routing Setup
+
+### params middleware
+
+- set req vars
+  - **setDateRange**
+    - req.period
+    - req.dateRange
+- redirect
+
+### shared middleware
+
+- **saveUrl**
+
+### routing middleware
+
+#### types
+
+- redirects
+- simple routes
+- routes with form-wizard
+- sub-routers
+- arrays of grouped middleware
+  - collections middleware (`setActions`,`setContext`,`setPagination`)
+  - data setup, retrieval and transformation middleware (`setBody` `setFilter` `setResults`)
+
+#### router.`HTTP_METHOD`
+
+##### Generic middleware
+
+- **path/with/:tokens**
+- **protectRoute**
+- redirects
+
+##### Collections setup
+
+- setActions
+- setContext
+  - permission
+  - text
+  - href
+- setPagination
+
+##### Data setup, retrieval and transformation
+
+- setBody*Something*
+  - sets `req.body.something`
+    - structure includes `status`, `sortBy`, `sortDirection`, `[locationType]`
+- set*Something*Filters
+  - calls services
+  - uses body prop
+  - sets `req.filter`
+- set*Something*Results
+  - calls services
+  - uses body prop
+  - sets `req.resultsSomething` or `req.resultsSomethingAsTable`
+
+##### Render
+- `render` method from controller, e.g. `home/controllers.js:dashboard`
+  - `res.render` with template e.g. `home/dashboard`
+    - actions
+    - context
+    - dateRange
+    - filter
+    - pagination
+    - period
+    - resultsVars
+
+# Questions
+
+- what is the difference between setBody, setFilter and setResults
```

Comment:
Did I manage to answer this ok on Slack the other day?

Reply:
This is the current structure:

`setBody` — used to build the shared body sent to the service calls to generate the filter and the results. Contains a shared query for date filters, location filters, etc. Attached result to `req.body[type]`

`setFilter` — builds the filter component for that collection. Uses `req.body[type]` as a base and overrides the filter based on the desired counts to retrieve, usually status. Attached the result as a filter component to `req.filter`.

`setResults` — makes a call to the relevant service method to return the results that should be displayed for the current view. Uses `req.body[type]` which is based on the current params and filters in the URL. Also responsible for calling the relevant presenter for this resource type and sets the results as `req.resultsAsTable` or `req.resultsAsCards` to be used by the controller/template.

Reply:
Reworked this section based upon slack / other PR comments